libraries/standard/mqtt/src/mqtt.c

```diff
@@ -965,9 +988,17 @@ static MQTTStatus_t receiveSingleIteration( MQTTContext_t * pContext,
     }
     else
     {
-        /* Receive packet. Remaining time is recalculated before calling this
-         * function. */
-        status = receivePacket( pContext, incomingPacket, remainingTimeMs );
+        if( pContext->networkBuffer.pBuffer != NULL )
```

Comment:
do we need to check it here as we have already asserted this in line 563?

Reply:
Yes because the pattern of assert() is for developer errors. It is to follow a function where the condition has been checked previously outside of the assert.

Reply:
I realized github hid the lines and I was taking a look at a wrong line number.

Reply:
Per offline conversation we have decided to move this check to the public function scope.